src/colorizer/utils/dataset_utils.ts

```diff
@@ -0,0 +1,92 @@
+// Defines types for working with dataset manifests, and methods for
+// updating manifests from one version to another.
+
+export type ManifestFileMetadata = {
+  /** Dimensions of the frame, in scale units. Default width and height are 0. */
+  frameDims: {
+    width: number;
+    height: number;
+    units: string;
+  };
+  frameDurationSeconds: number;
+  /* Optional offset for the timestamp. */
+  startTimeSeconds: number;
+};
+
+type ManifestFileV1 = {
+  frames: string[];
+  /** Map from feature name to relative path */
+  features: Record<string, string>;
+  /** Deprecated; avoid using in new datasets. Instead, use the new `FeatureMetadata` spec. */
+  featureMetadata?: Record<
+    string,
+    Partial<{
+      units?: string | null;
+      type?: string | null;
+      categories?: string[] | null;
+    }>
+  >;
+  outliers?: string;
+  tracks?: string;
+  times?: string;
+  centroids?: string;
+  bounds?: string;
+  metadata?: Partial<ManifestFileMetadata>;
+};
+
+// V2 removes the featureMetadata field, replaces the features map with an ordered
+// array of metadata objects.
+type ManifestFileV2 = Omit<ManifestFileV1, "features" | "featureMetadata"> & {
+  features: {
+    name: string;
+    data: string;
+    units?: string;
+    type?: string;
+    categories?: string[];
+  }[];
+};
```

Comment:
I think I'm going to keep it as an inline type for now because I think it's more readable like this? But if another version gets added that continues to change `features`, it'll probably be worth moving to its own type. Thinking through this was a good exercise, thank you!